### ISSUE TITLE
RHost build fails with latest R installation

### ISSUE THREAD

**Issue by karthiknadig** [Microsoft/RTVS#3619]
The registry location has changed and current detection for R during build in CMakeLists.txt is not sufficient. I get the error "R 3.4.x not found" while building.
![image](https://user-images.githubusercontent.com/3840081/27069612-7532e4a2-4fca-11e7-97ca-8096ae27d1cf.png)

```
    foreach(i RANGE 20 0 -1)
        if("${R_PATH}" STREQUAL "/registry")
            GET_FILENAME_COMPONENT(R_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\R-core\\R64\\3.4.${i};InstallPath]" ABSOLUTE)
        endif()
    endforeach()
```

**Comment by int19h** [comment]
Is this for MRO specifically? It works fine for me for CRAN R 3.4.0.

**Comment by karthiknadig** [comment]
Yes, I realized this was for MRO only after you mentioned. The error message needs to be specific, or we should support building with MRO.

**Comment by int19h** [comment]
Going forward, we'll likely switch to using our own header anyway, or perhaps using checked-in headers from various R versions - using the system headers is not going to be workable long-term with various incompatible versions. So I wouldn't worry much about this.
